src/server/controllers/courses.js

```diff
@@ -57,4 +57,21 @@ exports.details= async (req, res) => {
   } catch (error) {
     res.send({ error });
   }
-};
\ No newline at end of file
+};
+
+exports.post = async (req, res) => {
+  try {
+    const { courseData } = req.body;
+    const { count } = await courses.findAndCountAll({
+      where: {
+        course_name: courseData.course_name
+      }
+    });
+    if (count !== 0) throw new TypeError("The name is used before");
```

Comment:
It might be cleaner to send a 400 status code with an error message here instead of throwing a TypeError.  I think the express default error handler will catch that TypeError exception and turn it into a http 500 response, but 400 would be more correct for this situation: It's not really a server-side problem, it's a client-side problem - the user passed bad data.  A 5xx error implies that the client should feel free to retry the same request and it might work next time, but a 4xx implies that the client did something wrong and needs to fix their request in some way (which I think is what you want here).

Reply:
I will do it in another branch for all of files